hazelcast/src/main/java/com/hazelcast/map/impl/MapContainer.java

```diff
@@ -108,7 +116,15 @@ public MapContainer(final String name, final Config config, final MapServiceCont
         this.objectNamespace = new DistributedObjectNamespace(MapService.SERVICE_NAME, name);
         initWanReplication(nodeEngine);
         this.extractors = new Extractors(mapConfig.getMapAttributeConfigs(), config.getClassLoader());
-        this.indexes = new Indexes((InternalSerializationService) serializationService, extractors);
+        if (!mapConfig.getInMemoryFormat().equals(NATIVE)) {
```

Comment:
It's a non-native case (there's a not operator there), so I need this check here.
I removed the else clause, that was doing nothing anyway, so it's less confusing.